modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/snapshot/FullStateTransferIndexChooser.java

```diff
@@ -274,30 +267,58 @@ private long catalogActivationTimestampBusy(int catalogVersion) {
         return catalog.time();
     }
 
-    private void recoveryReadOnlyIndexesBusy() {
+    private void recoverReadOnlyIndexesBusy() {
         int earliestCatalogVersion = catalogService.earliestCatalogVersion();
         int latestCatalogVersion = catalogService.latestCatalogVersion();
 
-        // At the moment, we will only use tables from the latest version (not dropped), since so far only replicas for them are started
-        // on the node.
-        int[] tableIds = catalogService.tables(latestCatalogVersion).stream().mapToInt(CatalogObjectDescriptor::id).toArray();
-
-        Map<Integer, ReadOnlyIndexInfo> readOnlyIndexes = new HashMap<>();
+        var readOnlyIndexById = new HashMap<Integer, ReadOnlyIndexInfo>();
+        var previousCatalogVersionTableIds = Set.<Integer>of();
 
         for (int catalogVersion = earliestCatalogVersion; catalogVersion <= latestCatalogVersion; catalogVersion++) {
-            Catalog catalog = catalogService.catalog(catalogVersion);
+            long activationTs = catalogActivationTimestampBusy(catalogVersion);
 
-            assert catalog != null : catalogVersion;
+            catalogService.indexes(catalogVersion).stream()
+                    .filter(index -> index.status() == STOPPING)
+                    .forEach(index -> readOnlyIndexById.computeIfAbsent(index.id(), i -> new ReadOnlyIndexInfo(index, activationTs)));
 
-            for (int tableId : tableIds) {
-                for (CatalogIndexDescriptor index : catalog.indexes(tableId)) {
-                    if (index.status() == STOPPING) {
-                        readOnlyIndexes.computeIfAbsent(index.id(), indexId -> new ReadOnlyIndexInfo(tableId, catalog.time(), indexId));
-                    }
-                }
+            Set<Integer> currentCatalogVersionTableIds = tableIds(catalogVersion);
+
+            int finalCatalogVersion = catalogVersion;
```

Comment:
Please add a comment explaining that here we look for indices that transitioned directly from AVAILABLE to [deleted] (corresponding to the logical READ_ONLY state) as such transitions only happen when a table is dropped.

Reply:
adds